packages/js/components/src/form/form-context.ts

```diff
@@ -23,15 +40,10 @@ export type FormContext< Values extends Record< string, any > > = {
 	handleSubmit: () => Promise< Values >;
 	getInputProps< Value extends Values[ keyof Values ] >(
 		name: string
-	): {
-		value: Value;
-		checked: boolean;
-		selected?: boolean;
-		onChange: ( value: ChangeEvent< HTMLInputElement > | Value ) => void;
-		onBlur: () => void;
-		className: string | undefined;
-		help: string | null | undefined;
-	};
+	): InputProps< Value >;
+	getDateTimePickerControlProps< Value extends Values[ keyof Values ] >(
+		name: string
+	): DateTimePickerControlProps< Value >;
```

Comment:
I am not a huge fan of adding a specific `DateTimePickerControlProps` function that is very specific to the use case of our `DateTimePicker` component. I think we should keep this relatively generic.

I know @octaedro ran into this also with select controls and actually created some util functions to help with this: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/products/sections/utils.ts#L36

So I do think we should leave this up mostly to the consumer.
An alternative option to solve this and keep it generic, is to provide a second option to the `getInputProps` function with a `props` list, if not defined we return all of them, if set, we only return the specific list.

Something like: `getInputProps( 'date', ['onChange', 'onBlur', 'className', 'help'] )`

We could also switch it, and only do the `omit` list if that is shorter by default.

Reply:
Hmm... I've never been a fan of the `getInputProps` as a generic function, but I think I'm getting too hung up on the naming, and equating the "input" in its name with an HTML `<input>` element, not a more generic form "input field".

I'm also not a fan of our use of the `utils` dumping grounds (and, that specific `utils` is not generic to `Form`, but specific to our products section), but that's another topic. 😄

I think we should provide support in `Form` (and therefore `getInputProps`) to help consumers out, so they can easily add components to a form without having to do a lot of unnecessary work.

Both you and @mdperez86 have expressed a dislike of adding a specific `getDateTimePickerControlProps` function. And, I am coming around to your viewpoint now on this. 

I like the idea of providing an optional props list to limit what props get returned. In addition, for this specific case, I need to "map" the `value` prop to the `currentDate` prop... so, perhaps an optional prop mapping could be supplied, to help consumers easily adapt their component to work with `getInputProps`?

Or, in this specific case, since we control both the component and Form, I could just change `DateTimePickerControl`'s `currentDate` prop to be named `value`, but other consumers might not have that flexibility.

Or, I could write a thin wrapper component for `DateTimePickerControl` to adapt it, but that seems heavy-handed.

You know, the more I write and the more I think about this, I think you are right... we should leave this totally to the consumer. I think I will most of these changes out of this PR, keeping just the addition of the `InputProps` type, which I think is a nice code readability and maintainability improvement, and a tweaked Storybook story with a DateTimePickerControl on the form.